docs/ThemeAggregation.md

```diff
@@ -0,0 +1,82 @@
+# Theme Aggregation
+
+Terra Toolkit provides a built-in mechanism for aggregating themes.
+
+By default, the terra-toolkit webpack configuration enables theme aggregation when it detects the presence of a `terra-theme.config.js` file.
+
+## Getting started
+
+Create a new file within the same directory as your webpack configuration file.
+
+The file must be named `terra-theme.config.js`.
+
+```
+project
+├── terra-theme.config.js
+├── webpack.config.js
+```
+
+Within this file declare and export an object containing your theme configuration.
+
+This configuration will be used to aggregate nested dependency themes and output a single `aggregated-themes.js` file. This file will be automatically included as an entry point within your application if you are using the webpack configuration provided by terra-toolkit.
+
+Theme files must follow naming conventions to be aggregated. Theme files are expected be within a namespaced directory within a `themes` directory.
+
+```txt
+project
+└── themes
+    ├── terra-dark-theme
+    │   ├── component.scss
+    │   ├── root-theme.scss
+    │   └── scoped-theme.scss
+    └── terra-light-theme
+        ├── component.scss
+        ├── root-theme.scss
+        └── scoped-theme.scss
+```
+
+### Root Theme Example
+
+The `root-theme.scss` file should include all the necessary imports for the theme.
+
+```scss
+@import 'component-1.scss';
+@import 'component-2.scss';
+```
+
+### Scoped Theme Example
+
+The `scoped-theme.scss` file should include a selector to enable scoping.
+
+```scss
+$selector: '.terra-dark-theme';
```

Comment:
The `$selector` variable is an implementation detail and assumes you have a Sass mixin within the component theme file that either outputs `:root` or the string defined in the `$selector`.

Adding a selector variable in this file won't make this a scoped theme unless the imported theme code uses a mixin to swap its contents between `:root` selector and `$selector` as a selector.

Should we document this? I'm thinking we should but this doesn't seem like the right place as it seems more that we need documentation on how to create a custom theme and that would be more fitting.

Reply:
I agree, we need some documentation on how to property create theme files. Any ideas where that should live? 

I'll remove this documentation for now so it doesn't confuse anyone.

Reply:
I think we should add something to terra-ui.